### ISSUE TITLE
qt: Fetching price may lead to random SEGFAULT

### ISSUE THREAD

**Issue by alamshafil** [AvianNetwork/Avian#95]
When running mainnet/testnet/regtest there is a random chance that it may crash to a segfault.

**Comment by alamshafil** [comment]
Update: SEGFAULT was experienced on main-net. This means the issue is unrelated to assets or UI.

**Comment by alamshafil** [comment]
SEGFAULT occurs here:
```cpp
int unit = walletModel->getOptionsModel()->getDisplayUnit();
```

`getOptionsModel()` is returning null, causing the wallet to crash.

**Comment by alamshafil** [comment]
Fixed in #172
